### ISSUE TITLE
[TT-3690] API Ownership

### ISSUE THREAD

**Issue by sedkis** [TykTechnologies/tyk-operator#249]
## Expected Behavior

I want to be able to add API ownership to my API definitions.

Here's the JSON in my API definition:
```
  "user_group_owners": [
    "6010369dd6cd2b0001bb7ddb",
    "60103741d6cd2b0001e38435"
  ],
  "user_owners": []
```

In the UI:
![image](https://user-images.githubusercontent.com/8964621/105872776-c9487900-5fc8-11eb-9433-50651c9f6be2.png)

### Potential BDD Test Coverage

```gherkin
Feature: <summary of feature>
  As a developer belonging to a team
  I want to add API Ownership of my team to an API definition
  So that teams cannot edit my API

  Scenario: <summary of a scenario within this feature>
    Given an API Definition
    When I add API Ownership to my API
    Then I want to get API ownership to my API
```

**Comment by joshblakeley** [comment]
Sorry if i'm being dumb but isn't the point of the operator that *IT* is the API owner and RBAC is then pushed into the kubernetes space by namespace access etc

**Comment by buger** [comment]
You still may need RBAC for the dashboard. Analytics, developers and etc. K8s handles only configurations.

**Comment by sedkis** [comment]
> Sorry if i'm being dumb but isn't the point of the operator that _IT_ is the API owner and RBAC is then pushed into the kubernetes space by namespace access etc

Yes, but @buger  is spot on:

```
As a developer belonging to a dev team
  I want to add API Ownership of my team to an API definition
  So that other teams cannot edit my API
```

**Comment by joshblakeley** [comment]
Great thanks for the clarification both!

**Comment by asoorm** [comment]
This means we need to create user group CRDs in K8s also right?

**Comment by patriziobrunops** [comment]
The main problem here is that ownership info is not part of the api definition. It's the dashboard API that uses and updates the ownership info. 

Api ownership support requires the `dashboard.DashboardApi` struct to change like as follows:
```go
type DashboardApi struct {
	CreatedAt       string                  `json:"created_at,omitempty"`
	ApiDefinition   model.APIDefinitionSpec `json:"api_definition"`
        UserOwners      []string                `json:"user_owners,omitempty"`
        UserGroupOwners [] string               `json:"user_group_owners,omitempty"`
}
```

Since the `ApiDefinition.spec` doesn't include ownership info, this should be in a separate CRD.

**Comment by patriziobrunops** [comment]
@asoorm I implemented the api-ownership management using `OperatorContext`. I added `user_owners` and `user_group_owners` to `OperatorContext.spec`, they're string arrays expecting user/group IDs. I also added two new keys to secret `.spec.data`: `TYK_USER_OWNERS` and `TYK_USER_GROUP_OWNERS`, they're expected to contain comma separated lists of user/group IDs. To you mind if I submit a PR?

**Comment by buraksekili** [comment]
it is implemented in #483 and is now available with version Tyk Operator v0.12.0.
